liquibase-core/src/main/java/liquibase/datatype/core/TimestampType.java

```diff
@@ -16,14 +16,12 @@
 import liquibase.util.StringUtils;
 import liquibase.util.grammar.ParseException;
 
-import java.util.Locale;
-
 /**
  * Data type support for TIMESTAMP data types in various DBMS. All DBMS are at least expected to support the
  * year, month, day, hour, minute and second parts. Optionally, fractional seconds and time zone information can be
  * specified as well.
  */
-@DataTypeInfo(name = "timestamp", aliases = {"java.sql.Types.TIMESTAMP", "java.sql.Timestamp", "timestamptz"}, minParameters = 0, maxParameters = 1, priority = LiquibaseDataType.PRIORITY_DEFAULT)
+@DataTypeInfo(name = "timestamp", aliases = {"java.sql.Types.TIMESTAMP", "java.sql.Types.TIMESTAMP_WITH_TIMEZONE", "java.sql.Timestamp", "timestamptz"}, minParameters = 0, maxParameters = 1, priority = LiquibaseDataType.PRIORITY_DEFAULT)
```

Comment:
In JDBC the `java.time.OffsetDateTime` is used for `Types.TIMESTAMP_WITH_TIMEZONE`.

Reply:
Why is that relevant here? The `Timestamp` type in Liquibase represents a TIMESTAMP in SQL, with or without a Timezone. The code in this class is used for creating a database schema, not reading data out of the table. Or am I missing something?

Reply:
`TIMESTAMP [ WITHOUT TIME ZONE ]` and `TIMESTAMP WITH TIME ZONE` should not have the same data type, they are different. The only one database that uses the similar internal representation for them is PostgreSQL and it looks like the violation of the SQL Standard by this database.

AFAIK, other databases not directly related with this one really hold the time zone offset component (Oracle also can hold the name of the time zone, that is not supported by the Standard). Legacy, buggy and defective by design `java.sql.Timestamp` is not fully reliable for the plain `TIMESTAMP`, but it is also completely unsuitable for `TIMESTAMP WITH TIME ZONE` for all databases with exception for PostgreSQL, it can only lead to data loss.

I think you need to request a review from somebody from Luquibase community. Such changes should be carefully designed.

Reply:
> The code in this class is used for creating a database schema, not reading data out of the table. Or am I missing something?

If it is not really used anywhere it could be OK, but in that case you also need to check for `TIMESTAMPTZ` from PostgreSQL below
```Java
        if (originalDefinition.startsWith("java.sql.Types.TIMESTAMP_WITH_TIMEZONE")
            && (database instanceof PostgresDatabase
            || database instanceof OracleDatabase
            || database instanceof H2Database
            || database instanceof HsqlDatabase)) {
```

Reply:
Or maybe not and `TIMESTAMPTZ` will be used as is. Such things should be tested somewhere.

Reply:
This code is about creating timestamps in a database schema. It has nothing to do with the representation of that timestamp in Java code. If I wanted to store a timestamp with a timezone in a **Java** class, I would use the `java.time.OffsetDateTime` type, exactly as you suggest. But I would map that field to a `TIMESTAMP WITH TIMEZONE` **SQL** type. This is the [approach used in Hibernate](https://docs.jboss.org/hibernate/orm/5.4/userguide/html_single/Hibernate_User_Guide.html#basic-datetime-java8).

The Liquibase docs state that ['..specifying a java.sql.Types.* type will be converted to the correct type..'](https://www.liquibase.org/documentation/column.html). However, `java.sql.Types.TIMESTAMP_WITH_TIMEZONE` does not work correctly, which is the point of this PR. I tried this approach the first time I wanted to create a `TIMESTAMP WITH TIMEZONE` SQL type. From studying the Liquibase code, I now know I could have used the raw SQL type `TIMESTAMP WITH TIMEZONE` instead. I created this PR to avoid others falling into the same trap. If support for this type isn't added, the Liquibase docs should be updated to reflect that position.